esmvaltool/cmorizers/data/formatters/datasets/tropflux.py

```diff
@@ -0,0 +1,146 @@
+# pylint: disable=unused-argument
+# pylint: disable=too-many-arguments
+# pylint: disable=too-many-function-args
+# pylint: disable=R0917
+# pylint: disable=E1121
+# flake8: noqa
+"""ESMValTool CMORizer for TROPFLUX data.
+
+Tier
+    Tier 2: other freely-available dataset.
+
+Source
+    https://incois.gov.in/tropflux
+
+Last access
+    20250115
+
+Download and processing instructions
+    Register and sign-in here:
+
+    https://incois.gov.in/tropflux/DataHome.jsp
+
+    Data is available per year from 1979 to 2018
+    at daily and monthly frequency.
+
+    Products availables are:
+
+    - Air Temperature at 2m
+    - Latent Heat Flux
+    - Long Wave Radiation
+    - Meridional Wind Stress
+    - Net Surface Heat Flux
+    - Sea Surface Temperature
+    - Sensible Heat Flux
+    - Shortwave Radiation
+    - Specific Humidity at 2m
+    - Wind Speed at 10m
+    - Wind Stress magnitude
+    - Zonal Wind Stress
+
+Caveats
+"""
+
+import logging
+import re
+from copy import deepcopy
+from pathlib import Path
+
+import cf_units
+import iris
+from esmvalcore.cmor.table import CMOR_TABLES
+from iris import NameConstraint
+from iris.util import equalise_attributes
+
+from esmvaltool.cmorizers.data import utilities as utils
+
+try:
+    iris.FUTURE.date_microseconds = True
+    iris.FUTURE.save_split_attrs = True
+except AttributeError as e:
+    # Handle cases where FUTURE or the attributes don't exist
+    print(f"AttributeError: {e}")
+except (TypeError, ValueError) as e:
+    # Handle specific errors if these might occur
+    print(f"TypeError or ValueError: {e}")
+except BaseException as e:
+    # Fallback for rare or unknown issues, but avoid catching Exception
+    print(f"An unexpected error occurred: {e}")
+
+logger = logging.getLogger(__name__)
```

Comment:
```suggestion

logger = logging.getLogger(__name__)

try:
    iris.FUTURE.date_microseconds = True
    iris.FUTURE.save_split_attrs = True
except AttributeError as e:
    # Handle cases where FUTURE or the attributes don't exist
    logger.warning(f"AttributeError: {e}")
except (TypeError, ValueError) as e:
    # Handle specific errors if these might occur
    logger.warning(f"TypeError or ValueError: {e}")
except BaseException as e:
    # Fallback for rare or unknown issues, but avoid catching Exception
    logger.warning(f"An unexpected error occurred: {e}")
```
sending messages to the log file instead of print. (I realised my iris was an older version)

Reply:
Very good point